### PR TITLE
Support distributed transactions across multiple logical database(#19…

### LINKED ISSUE #19894: The data source is messed up.
## Bug Report
The data source is messed up

### Which version of ShardingSphere did you use?
master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Actual behavior
Data that should have been inserted into another data source is inserted into another data source.

### Reason analyze (If you can)
In a transaction, different logical libraries are operated, and the wrong data source is obtained due to the existence of cachedConnections.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
```SQL
mysql> begin;
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
Connection id:    1
Current database: *** NONE ***

Query OK, 0 rows affected (0.61 sec)

mysql> insert into sharding_blob.t_order_0 values(132, 48,12,3);
Query OK, 1 row affected (0.22 sec)

mysql> insert into sharding_db2.t_order values(122, 48,12,3);
```

`values(122, 48,12,3);` need insert into `xx.xx.xx.34:3366`, but insert into `xx.xx.49:3306`;

### Example codes for reproduce this issue (such as a github link).
```yaml
databaseName: sharding_blob

dataSources:
  ds_0:
    url: jdbc:mysql://xx.xx.49:3306/demo_ds_0?serverTimezone=UTC&useSSL=false
    username: test
    password: 123456
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
    minPoolSize: 1
```

```yaml
databaseName: sharding_db2

dataSources:
  ds_0:
    url: jdbc:mysql://xx.xx.xx.34:3366/demo_ds_0?serverTimezone=UTC&useSSL=false
    username: test
    password: 123456
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
    minPoolSize: 1
  ds_1:
    url: jdbc:mysql://xx.xx.xx.49:3306/demo_ds_1?serverTimezone=UTC&useSSL=false
    username: test
    password: 123456
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
    minPoolSize: 1

rules:
  - !SHARDING
    broadcastTables:
      - t_order_9999
    tables:
      t_order:
        actualDataNodes: ds_${0..1}.t_order_${0..1}
        tableStrategy:
          standard:
            shardingColumn: order_id
            shardingAlgorithmName: t_order_inline
        keyGenerateStrategy:
          column: order_id
          keyGeneratorName: snowflake
      t_order_item:
        actualDataNodes: ds_${0..1}.t_order_item_${0..1}
        tableStrategy:
          standard:
            shardingColumn: order_id
            shardingAlgorithmName: t_order_item_inline
        keyGenerateStrategy:
          column: order_item_id
          keyGeneratorName: snowflake
    defaultDatabaseStrategy:
      standard:
        shardingColumn: user_id
        shardingAlgorithmName: database_inline
    defaultTableStrategy:
      none:
    shardingAlgorithms:
      database_inline:
        type: INLINE
        props:
          algorithm-expression: ds_${user_id % 2}
          allow-range-query-with-inline-sharding: true
      t_order_inline:
        type: INLINE
        props:
          algorithm-expression: t_order_${order_id % 2}
          allow-range-query-with-inline-sharding: true
      t_order_item_inline:
        type: INLINE
        props:
          algorithm-expression: t_order_item_${order_id % 2}
          allow-range-query-with-inline-sharding: true

    keyGenerators:
      snowflake:
        type: SNOWFLAKE

```

### PR DESCRIPTION
**Pull request by FlyingZC**
Fixes #19894.

Changes proposed in this pull request:
- Support XA and BASE distributed transactions across multiple logical database
- Allow `use xx_db` in transaction

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
@FlyingZC Please fix ci error first.

**Comment by FlyingZC** [comment]
> @FlyingZC Please fix ci error first.

I will check.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ResourceDataSource.java [hunk 1]
```diff
@@ -34,8 +35,10 @@ public final class ResourceDataSource {
     private final DataSource dataSource;
     
     public ResourceDataSource(final String originalName, final DataSource dataSource) {
+        final String[] databaseAndDataSourceName = originalName.split("\\.");
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ResourceDataSource.java:38]: Please remove final here.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ConnectionTransaction.java [hunk 2]
```diff
@@ -46,6 +48,7 @@ public ConnectionTransaction(final String databaseName, final TransactionRule ru
     }
     
     public ConnectionTransaction(final String databaseName, final TransactionType transactionType, final TransactionRule rule) {
+        this.databaseName = databaseName;
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ConnectionTransaction.java:51]: Should databaseName be extracted from SQL?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ConnectionTransaction.java (hunk 2 above)
Comment (FlyingZC) [reply shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/ConnectionTransaction.java:51]: I will check.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ResourceDataSource.java (hunk 1 above)
Comment (FlyingZC) [reply shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-core/src/main/java/org/apache/shardingsphere/transaction/core/ResourceDataSource.java:38]: You ar right.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**